plugins/speculation-rules/readme.txt

```diff
@@ -88,6 +88,16 @@ add_filter(
 
 As mentioned above, adding the `no-prerender` CSS class to a link will prevent it from being prerendered (but not prefetched). Additionally, links with `rel=nofollow` will neither be prefetched nor prerendered because some plugins add this to non-idempotent links (e.g. add to cart); such links ideally should rather be buttons which trigger a POST request or at least they should use `wp_nonce_url()`.
 
+= How will this impact analytics and personalization? =
+
+Prerendering can affect analytics and personalization.
+
+For client-side JavaScript, is recommended to delay these until the page clicks and some solutions (like Google Analytics) already do this automatically for prerender. See [Impact on Analytics](https://developer.chrome.com/docs/web-platform/prerender-pages#impact-on-analytics). Additionally, cross-origin iframes are not loaded until activation which can further avoid issues here.
+
+Speculating on hover (moderate) increases the chance the page will be loaded, over preloading without this signal, and thus reduces the risk here. Alternatively, the plugin offers to only speculate on mouse/pointer down (conservative) which further reduces the risk here and is an option for sites which are concerned about this, at the cost of having less of a lead time and so less of a performance gain.
+
+A prerendered page is linked to the page that prerenders it, so personalisation may already be known by this point and changes (e.g. browsing other products, or logging in/out) may require a new page load, and hence a new prerender anyway, which will take these into account. But it definitely is something to be aware of and test!
```

Comment:
What can be done if the test leads to discovering problems? I think something should be included about that, e.g. something about modifying the personalization logic or, if that's not possible/feasible, excluding the relevant pages from speculative loading (via the filter the plugin provides).

Reply:
The options (from best to worse IMHO) are:
- Cancel speculations on personalization changes (note this happens automatically on navigations anyway, which is probably when most personalization signal changes and other state changes happen anyway).
- Update on page activation with client-side JS (plus: also could handle stale background tabs, negative: can lead to a flash of old content(
- Don't speculation on hover and only do on mousedown / "conservative" eagerness. (negative, less gain).
- Exclude personalized pages (but if this is a loging widget for example that might be ALL pages)
- Don't use speculation rules on those types of sites.

Probably not a bad idea to spell these out.